lib/rufo/new_formatter.rb

```diff
@@ -266,13 +266,26 @@ def visit_begin(node)
 
   def visit_string_literal(node)
     # [:string_literal, [:string_content, exps]]
-    consume_token :on_tstring_beg
-    
-    inner = node[1..-1]
-    
+    if current_token_kind == :on_backtick
+      consume_token :on_backtick
+    else
+      consume_token :on_tstring_beg
+    end
+
+    visit_string_literal_end(node)
+  end
+
+  def visit_string_literal_end(node)
+    inner = node[1]
+    inner = inner[1..-1] unless node[0] == :xstring_literal
     visit_exps(inner, with_lines: false)
 
-    consume_token :on_tstring_end
+    case current_token_kind
```

Comment:
Can we make this conditional and the conditional in `visit_string_literal` the same shape? Either both use `case` or both use `if`?